### PR TITLE
Travis cleanup

### PR DESCRIPTION
**Pull request by gberaudo**

### CODE REVIEW

**Comment by gberaudo** [comment]
@sbrunner, the build is passing. Could you please have a look to check if it is all good?

File: buildtools/cleanup-ghpages.py [hunk 1]
```diff
@@ -5,21 +5,21 @@
...
+        json = requests.get(url).json()
         expected = [
-            branch["name"] for branch in loads(requests.get(url).content)
+            branch["name"] for branch in json
         ]
         expected.append("index.html")
         expected.append(".git")
         for path in listdir(sys.argv[2]):
-            if path not in expected:
+            if path not in expected or path.startswith("greenkeeper/"):
```
Comment (gberaudo) [inline buildtools/cleanup-ghpages.py]: to remove

Comment (gberaudo) [reply buildtools/cleanup-ghpages.py]: and do not add when the branch has a /

File: buildtools/deploy.sh [hunk 2]
```diff
@@ -1,90 +1,57 @@
...
+# Some actions started from the root directory
+# publish-examples
+TARGET="$TMP/${GIT_BRANCH}/examples/"
+mkdir -p $TARGET
+cp -r .build/examples-hosted/* $TARGET
+# publish-apps-gmf
+TARGET="$TMP/${GIT_BRANCH}/examples/contribs/gmf/"
+mkdir -p $TARGET
+cp -r .build/examples-hosted/contribs/gmf/apps $TARGET
+cp -r .build/examples-hosted/contribs/gmf/build $TARGET
```
Comment (gberaudo) [inline buildtools/deploy.sh]: to simplify

**Comment by gberaudo** [comment]
@sbrunner, done.

File: buildtools/deploy.sh [hunk 3]
```diff
@@ -1,90 +1,36 @@
...
-cd .build/ngeo-${GITHUB_USERNAME}-gh-pages/
+# Cleanup the git tree by creating a single commit with all content
+pushd $TMP
 git add -A
 git status
-git commit -m 'Update GitHub pages' || true
-git push ${GIT_REMOTE_NAME} gh-pages
+FIRST_COMMIT=$(git log --format='%H' | tail -1)
+git reset --mixed $FIRST_COMMIT
+git commit -a -m 'Latest GitHub pages'
```
Comment (sbrunner) [inline buildtools/deploy.sh]: Why not `git commit --amend` to have just one commit?

File: buildtools/deploy.sh [hunk 4]
```diff
@@ -1,90 +1,36 @@
...
 
-cd .build/ngeo-${GITHUB_USERNAME}-gh-pages/
+# Cleanup the git tree by creating a single commit with all content
+pushd $TMP
 git add -A
 git status
-git commit -m 'Update GitHub pages' || true
-git push ${GIT_REMOTE_NAME} gh-pages
+FIRST_COMMIT=$(git log --format='%H' | tail -1)
+git reset --mixed $FIRST_COMMIT
```
Comment (sbrunner) [inline buildtools/deploy.sh]: Should be before the `git add -A`

**Pull request merged**